crates/fuel-core/src/graphql_api/database.rs

```diff
@@ -252,6 +273,13 @@ impl ReadView {
         direction: IterDirection,
     ) -> impl Stream<Item = StorageResult<Message>> + '_ {
         futures::stream::iter(self.on_chain.all_messages(start_message_id, direction))
+            .chunks(self.batch_size)
+            .filter_map(|chunk| async move {
+                // Give a chance to other tasks to run.
+                tokio::task::yield_now().await;
+                Some(futures::stream::iter(chunk))
+            })
```

Comment:
Why not just `map`?

Reply:
I've pulled the code and I see this is the way to make it compile. It's annoying and adds misdirection though.

Reply:
Yes, only `filter_map` allows you to resolve the future=)

Reply:
Can we just write an extension trait for these spots? :P So we actually have a method that declares what it's doing rather than happening implicitly?

Reply:
i.e. roughly:
```rs
trait ResolveStreamFutures {
    type Item;
    fn resolve_each(&self) -> Item;
}
```

and then do a blanket impl of that for `Stream` or `stream::Iter` or whatever.

Reply:
I think `ResolveStreamFutures` is a bad name, we resolve nothing here, and stream itself already resolves futures=)

Reply:
Yes, it could have a better name. I was just putting something down. 

I don't want to have to write a whole alternative to `FilterMap` right now. It's kinda involved. So I'll just let it be. It's hard to understand though.

Reply:
https://github.com/FuelLabs/fuel-core/pull/2340/commits/727b2da6e460dc2fdb03e56ac01bef7f04f4379d